### ISSUE TITLE
Non-holomorphic networks with complex parameters not treated correctly

### ISSUE THREAD

**Issue by RehMoritz** [markusschmitt/vmc_jax#45]
If we define the class

```
class MatrixMultiplication_NonHolomorphic(nn.Module):
    holo: bool = False

    @nn.compact
    def __call__(self, s):
        layer1 = nn.Dense(1, use_bias=False, **init_fn_args(dtype=global_defs.tCpx))
        out = layer1(2 * s.ravel() - 1)
        if not self.holo:
            out = out + 1e-1 * jnp.real(out)
        return jnp.sum(out)
```

and let `holo=True`, the gradients computed by `psi.gradients` for the input `s=[0, 0, 0, 0]` are
`[-1.+0.j -1.+0.j -1.+0.j -1.+0.j -0.-1.j -0.-1.j -0.-1.j -0.-1.j]` as expected.

However, if we let `holo=False`, the returned gradients are `[-2.1+0.j -2.1+0.j -2.1+0.j -2.1+0.j]`. This means, that if we for example did time evolution with this setup, considering only the imaginary part of the S-matrix we would get all-zeroes.

If we add the gradient function

```
def flat_gradient_cpx_nonholo(fun, params, arg):
    gr = grad(lambda p, y: jnp.real(fun.apply(p, y)))(params, arg)["params"]
    gi = grad(lambda p, y: jnp.imag(fun.apply(p, y)))(params, arg)["params"]
    g = tree_flatten(tree_map(lambda x, y: [x.ravel(), -y.ravel()], gr, gi))[0]
    return jnp.concatenate(g)
```

the returned gradients are `[-1.1+0.j -1.1+0.j -1.1+0.j -1.1+0.j -0. -1.j -0. -1.j -0. -1.j -0. -1.j]`, which is in line with the above case where `holo=True`.
